R/split_functional_groups.R

```diff
@@ -0,0 +1,87 @@
+utils::globalVariables(c("functional_group", "species", "group"))
+#' Split the functional groups into species and group
+#'
+#' Split strings containing functional groups into species and group names.
+#' Functional groups can contain age ranges, plus groups, or sub categories,
+#' e.g., juvenile. This information is extracted from the functional group name
+#' and returned as the group name.
+#'
+#' @param x A character vector of functional group names. This is most often
+#'   created by running [unique()] on the functional group column of the data.
+#' @return
+#' A tibble with the following three columns: functional_group, species, and
+#' group, is returned.
+#' @export
+#' @author Kelli F. Johnson
+#' @examples
+#' # A hypothetical example of functional groups
+#' split_functional_groups(
+#'   c("spiny dogfish 0", "spiny dogfish 1-2", "spiny dogfish +")
+#' )
+#' split_functional_groups(c("spiny dogfish", "spiny dogfish juvenile"))
+#' data("ewe_nwatlantic_base", package = "ecosystemdata")
+#' split_functional_groups(unique(ewe_nwatlantic_base[["functional_group"]]))
+split_functional_groups <- function(x) {
+  if (!inherits(x, what = "character")) {
+    cli::cli_abort("{.var x} must be a character vector not a {class(x)}.")
+  }
+  groups <- unique(x) |>
+    gsub(pattern = "([a-z])([[:upper:]])", replacement = "\\1 \\2") |>
```

Comment:
Could you add a brief, plain-language comment explaining lines 29-31, similar to the one for lines 33-34?

Reply:
Thank you for the suggestion. Comments are now present.